domains/joinGame/JoinGameScreen.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from "react";
+import { ActivityIndicator, TextInput, View } from "react-native";
+import PropTypes from "prop-types";
+import PendingGames from "./PendingGames.jsx";
+
+const JoinGameScreen = ({ playerName }) => {
+    const [pendingGames, setPendingGames] = useState();
+    const [error, setError] = useState();
+    const [loadingPendingGamesState, setLoadingPendingGamesState] = useState(
+        "loading"
+    );
+    useEffect(() => {
+        if (!pendingGames) {
```

Comment:
you should test loadingPendingGamesState, also initilalize it to `pending`, and set it to loading only once you started to fetch